### DISCUSSION TITLE
How do i make packages that aren't compatible with pydantic v2 work

### DISCUSSION THREAD

**Discussion by yogesh1234567890** [fastapi/fastapi#10092] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
alembic==1.11.2
fastapi==0.101.1
uvicorn==0.23.2
fastapi_pagination==0.12.8
pandas==2.0.3
fastapi_utils==0.2.1
```

### Description

ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
fastapi-utils 0.2.1 requires pydantic<2.0,>=1.0, but you have pydantic 2.1.1 which is incompatible.

How do i make this work as some packages don't support pydantic v2

### Operating System

macOS

### FastAPI Version

0.101.1

### Pydantic Version

2.1.1

### Python Version

3.10.9

**Comment by n8sty** [comment, marked as answer]
Sadly, you don't. Or, well, you can, but it won't be fun. One option is to vendor the different dependencies you have that are conflicting with Pydantic. If the parts of those dependencies that you're using don't conflict with Pydantic 2.x, then import them as you would any other package. Or, if they do conflict then you'll have to modify the vendored dependency in some way.

Vendoring is the process of bringing a dependency into the source control of your project.

**Comment by vijaynp007** [comment]
fastapi-utils 0.2.1 requires pydantic<2.0,>=1.0, but you have pydantic 2.7.0 which is incompatible  How to resolve this installation issue  

unable to install fastapi-utils in pydantic 2.7.0

ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
fastapi-utils 0.2.1 requires pydantic<2.0,>=1.0, but you have pydantic 2.7.0 which is incompatible.

pls provide solution
